samza-core/src/main/java/org/apache/samza/container/LocalityManager.java

```diff
@@ -63,9 +64,11 @@ public LocalityManager(MetadataStore metadataStore) {
     metadataStore.all().forEach((containerId, valueBytes) -> {
         if (valueBytes != null) {
           String locationId = valueSerde.fromBytes(valueBytes);
-          Map<String, String> values = new HashMap<>();
-          values.put(SetContainerHostMapping.HOST_KEY, locationId);
-          allMappings.put(containerId, values);
+          if (StringUtils.isNotBlank(locationId)) {
```

Comment:
It appears that we're invoking StringUtils.isNotBlank immediately after <serde>.fromBytes(). 
Is it possible to create a wrapper for serde and call this method there. 

Then you check like so:
`
String locationId = new MySerdeWrapper(valueSerde).fromBytes(valueBytes);

public class MySerdeWrapper {
  public Deserializer deserializer;

  public MySerdeWrapper(Deserializer deserializer) {
    this.deserializer = deserializer;
  }

  public Optional<String> fromBytes(byte[] bytes) {
    return StringUtils.isNotBlank(deserializer.fromBytes(bytes)) ? Optional.of(deserializer.fromBytes(bytes)) : Optional.empty();
  }


}`

Reply:
I thought of doing this change before sending this patch out for review.  Since this is a critical bug-fix, I didn't want to couple it with a refactor/cleanup. If it's acceptable then I will do it in a follow-up [JIRA](https://issues.apache.org/jira/browse/SAMZA-2177).